spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/result/AbstractResult.java

```diff
@@ -68,4 +70,56 @@ public boolean failed() {
     public String getFailureDetail() {
         return failure == null ? null : String.join(", ", getFailureMessages());
     }
+
+    /**
+     * Executes a {@link Consumer} if this {@link Result} is successful
+     */
+    public void ifSuccess(Consumer<T> successAction) {
+        if (succeeded()) {
+            successAction.accept(getContent());
+        }
+    }
+
+    /**
+     * Executes a {@link Consumer} if this {@link Result} failed. Passes the {@link Failure} to the consumer
+     */
+    public void ifFailure(Consumer<F> failureAction) {
+        if (failed()) {
+            failureAction.accept(getFailure());
+        }
+    }
+
+    /**
+     * Executes a {@link Consumer} if this {@link Result} is successful. Returns this {@link Result} to enable fluent statements.
+     *
+     * @return this
+     */
+    public AbstractResult<T, F> whenSuccessful(Consumer<AbstractResult<T, F>> content) {
```

Comment:
This is the same as `ifSuccess`, because this pass the whole result, but the failure will be always null as it is successful.
Same for `orElseDo`, I would keep only the `ifSuccess`/`ifFailure` making them return `this` for their fluent use.

Reply:
the problem with `ifFailure` is a purely semantic one: we could then construct blocks like:
```java
res.ifSuccess(...)
   .ifFailure(...)
``` 
which (to my ears) sounds worse than
```java
res.ifSuccess(...)
   .orElse(...)
```
even if `ifFailure` and `orElse` perform the same function (maybe even call each other).

Reply:
they could be called `onSuccess` and `onFailure`